admin/tabs/roles/RolesModel.ts

```diff
@@ -0,0 +1,280 @@
+                    {name: 'effectiveRoleNames', displayName: 'Effective Roles', type: 'tags'},
+                    {name: 'members', type: 'json'}
+                ],
+                processRawData: raw => ({
+                    ...raw,
+                    effectiveUserNames: raw.effectiveUsers.map(it => it.name),
+                    effectiveDirectoryGroupNames: raw.effectiveDirectoryGroups.map(it => it.name),
+                    effectiveRoleNames: raw.effectiveRoles.map(it => it.name),
+                    inheritedRoleNames: raw.inheritedRoles.map(it => it.name)
+                })
+            },
+            colDefaults: {
+                filterable: true
+            },
+            columns: [
+                {field: {name: 'name', type: 'string'}},
+                {field: {name: 'category', type: 'string'}},
+                {field: {name: 'lastUpdated', type: 'date'}, ...Col.dateTime},
+                {field: {name: 'lastUpdatedBy', type: 'string'}},
+                {field: {name: 'notes', type: 'string'}, filterable: false, flex: 1}
+            ],
+            contextMenu: [...this.ACTIONS, '-', ...GridModel.defaultContextMenu]
+        });
+    }
+
+    private createRecordActions(): RecordActionSpec[] {
+        return [
+            {
+                text: 'Add',
+                icon: Icon.add(),
+                intent: 'success',
+                actionFn: () => this.createAsync()
+            },
+            {
+                text: 'Edit',
+                icon: Icon.edit(),
+                intent: 'primary',
+                actionFn: ({record}) => this.editAsync(record.data as HoistRole),
+                recordsRequired: true
+            },
+            {
+                text: 'Clone',
+                icon: Icon.copy(),
+                actionFn: ({record}) => this.createAsync(record.data as HoistRole),
+                recordsRequired: true
+            },
+            {
+                text: 'Delete',
+                icon: Icon.delete(),
+                intent: 'danger',
+                actionFn: ({record}) =>
+                    this.deleteAsync(record.data as HoistRole)
+                        .catchDefault()
+                        .linkTo(this.loadModel),
+                recordsRequired: true
+            }
+        ];
+    }
+
+    private createFilterChooserModel(): FilterChooserModel {
+        return new FilterChooserModel({
+            bind: this.gridModel.store,
+            fieldSpecs: [
+                'name',
+                'category',
+                'users',
+                'directoryGroups',
+                'roles',
+                'inheritedRoleNames',
+                'effectiveUserNames',
+                'effectiveDirectoryGroupNames',
+                'effectiveRoleNames',
+                'lastUpdatedBy',
+                {
+                    field: 'lastUpdated',
+                    example: 'YYYY-MM-DD',
+                    valueParser: (v, op) => {
```

Comment:
It looks like we have three copies of this thing in admin console now - could we DRY up, pending better support for date filtering?